src/com/xilinx/rapidwright/design/DesignTools.java

```diff
@@ -1028,24 +1028,77 @@ public static boolean removeConnectedRouting(Net net, Node node){
 	 * pins individually.
 	 * @param net The current net to modify routing and to which all pins will have their routing 
 	 * removed. If any pin passed in is not of this net, it is skipped and no effect is taken.
-	 * @param pins Pins that belong to the provided net that should have their selective routing 
-	 * removed.
+	 * @param pins Sink pins that belong to the provided net that should have their selective routing 
+	 * removed. This method only works for sink pins.  
+	 * See {@link #unrouteSourcePin(SitePinInst)} for handling source pin unroutes.
 	 */
 	public static void unroutePins(Net net, Collection<SitePinInst> pins) {
-	    Set<PIP> toRemove = getTrimmablePIPsFromPins(net, pins);
-	    ArrayList<PIP> updatedPIPs = new ArrayList<>();
-	    for(PIP pip : net.getPIPs()){
-	        if(!toRemove.contains(pip)) updatedPIPs.add(pip);
-	    }
-	    net.setPIPs(updatedPIPs);
+	    removePIPsFromNet(net,getTrimmablePIPsFromPins(net, pins));
 	    for(SitePinInst pin : pins) {
 	        pin.setRouted(false);
 	    }	    
 	}
-
+	
+	private static void removePIPsFromNet(Net net, Set<PIP> pipsToRemove) {
+	    if(pipsToRemove.size() > 0) {
+	        List<PIP> updatedPIPs = new ArrayList<>();
+	        for(PIP pip : net.getPIPs()){
+	            if(!pipsToRemove.contains(pip)) updatedPIPs.add(pip);
+	        }
+	        net.setPIPs(updatedPIPs);
+	    }	    
+	}
+	
+	/**
+	 * Unroutes a SitePinInst of a net.  This is desirable when a net has multiple SitePinInst
+	 * source pins (multiple outputs of a Site) and only a particular branch is desired to be 
+	 * unrouted.  If the entire net is to be unrouted, a more efficient method is {@link Net#unroute()}. 
```

Comment:
Would it be wise to do this automatically -- if the net only has one source pin (assuming the alternate source is populated correctly) then it can go ahead and unroute the whole net?

Reply:
In theory, yes we should.  But I'm wondering if we can always depend on the alternate source being populated correctly?  I guess we can make the change and find out, but perhaps we make that another PR.